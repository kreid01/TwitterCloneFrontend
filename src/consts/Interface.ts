export type IPost = {
    id:number,
    userName:string
    userAt:string
    userImg:string,
    postTextBody:string
    postMedia:string
    postDate:string
    commentCount:number
    likeCount:number
    retweetCount:number
    isLiked?:boolean
    isRetweeted?:boolean
    comments?: Array<Comment>
}

export type IComment = {
    userAt: string
    userName: string
    userImg: string 
    commentBody: string
    commentMedia?: string
    commentDate: string
}

export type INewComment = {
        commentBody: string;
        commentMedia: string;
        userAt: string;
        userName: string;
        userImg: string;
}