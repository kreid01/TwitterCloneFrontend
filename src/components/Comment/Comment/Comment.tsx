import React from "react";
import { Link } from "react-router-dom";
import { PostInteraction } from "../../PostInteractions/PostInteraction";

import { IComment } from "../../../consts/Interface";

interface Props {
  comment: IComment;
}

export const Comment: React.FC<Props> = ({ comment }) => {
  return (
    <div className="flex mt-8 mb-4 b border-b-slate-400">
      <img className="w-12 h-12 rounded-full" src={comment.userImg} alt="" />
      <div className="flex-col pl-4">
        <p className="">
          <Link to={`/${comment.userAt}`}>
            <strong>{comment.userName}</strong>
          </Link>
          <span className="text-gray-500">
            @{comment.userAt}Â·{comment.commentDate.substring(0, 7)}
          </span>
        </p>
        <p>{comment.commentBody}</p>
        <img
          className="w-80 pt-2 max-h-72 rounded-xl"
          src={comment.commentMedia}
          alt=""
          data-testid="media"
        />
      </div>
    </div>
  );
};
