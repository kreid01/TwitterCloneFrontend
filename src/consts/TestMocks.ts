export const testPost = {
  id: 1,
  userName: "test-name",
  userAt: "testAt",
  userImg: "testImg",
  postTextBody: "testBody",
  postMedia: "test-media",
  postDate: "testDate",
  commentCount: 1,
  likeCount: 1,
  retweetCount: 1,
  isRetweeted: false,
  isLiked: false,
  retweetedBy: [1],
  comments: [],
};

export const testComment = {
  id: 1,
  userName: "test-name",
  userAt: "testAt",
  userImg: "testImg",
  commentBody: "testBody",
  commentMedia: "test-media",
  commentDate: "test-date",
};
export const testPosts = [testPost, testPost];

export const testUser = {
  userId: 1,
  userName: "test-name",
  userAt: "testAt",
  userImg: "testImg",
  userCoverImg: "testCoverImg",
  userEmail: "testEmail",
  postsIds: [1, 2],
  joinDate: "testDate",
  following: [1, 2],
  followers: [1, 2],
  email: "",
  password: "",
};
