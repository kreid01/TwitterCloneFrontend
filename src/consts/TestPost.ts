export const testPost = {
    id:1,
    userName:'testName',
    userAt:'testAt',
    userImg:'testImg',
    postTextBody:'testBody',
    postMedia:'testMedia',
    postDate:'testDate',
    commentCount:1,
    likeCount:1,
    retweetCount:1
}