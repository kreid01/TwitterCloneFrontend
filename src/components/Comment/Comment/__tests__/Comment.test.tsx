/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { testComment } from "consts/TestMocks";
import { Comment } from "../Comment";
import { screen, render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

const mockFn = jest.fn;

afterEach(cleanup);
it("render without crashing", () => {
  render(
    <BrowserRouter>
      <Comment comment={testComment} />
    </BrowserRouter>
  );
});

it("post image renders", () => {
  render(
    <BrowserRouter>
      <Comment comment={testComment} />
    </BrowserRouter>
  );
  const image = screen.getByTestId<HTMLImageElement>("media");
  expect(image.src).toContain("test-media");
});

it("matched snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Comment comment={testComment} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
