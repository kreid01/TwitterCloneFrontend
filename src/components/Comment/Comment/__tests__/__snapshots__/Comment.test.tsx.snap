// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<div
  className="flex mt-8 mb-4 b border-b-slate-400"
>
  <img
    alt=""
    className="w-12 h-12 rounded-full"
    src="testImg"
  />
  <div
    className="flex-col pl-4"
  >
    <p
      className=""
    >
      <a
        href="/testAt"
        onClick={[Function]}
      >
        <strong>
          test-name
        </strong>
      </a>
      <span
        className="text-gray-500"
      >
        @
        testAt
        Â·
        test-da
      </span>
    </p>
    <p>
      testBody
    </p>
    <img
      alt=""
      className="w-80 pt-2 max-h-72 rounded-xl"
      data-testid="media"
      src="test-media"
    />
  </div>
</div>
`;
